src/serde.rs

```diff
@@ -0,0 +1,269 @@
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Disk", 7)?;
+
+        state.serialize_field("DiskType", &self.type_())?;
+        if let Some(s) = self.name().to_str() {
+            state.serialize_field("name", s)?;
+        }
+        state.serialize_field("file_system", &self.file_system())?;
+        state.serialize_field("mount_point", &self.mount_point())?;
+        state.serialize_field("total_space", &self.total_space())?;
+        state.serialize_field("available_space", &self.available_space())?;
+        state.serialize_field("is_removable", &self.is_removable())?;
+
+        state.end()
+    }
+}
+
+impl Serialize for dyn ProcessExt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Process", 18)?;
+
+        state.serialize_field("name", &self.name())?;
+        state.serialize_field("cmd", &self.cmd())?;
+        state.serialize_field("exe", &self.exe())?;
+        state.serialize_field("pid", &self.pid().as_u32())?;
+        state.serialize_field("environ", &self.environ())?;
+        state.serialize_field("cwd", &self.cwd())?;
+        state.serialize_field("root", &self.root())?;
+        state.serialize_field("memory", &self.memory())?;
+        state.serialize_field("virtual_memory", &self.virtual_memory())?;
+        if let Some(pid) = self.parent() {
+            state.serialize_field("parent", &pid.as_u32())?;
+        }
+        state.serialize_field("status", &self.status())?;
+        state.serialize_field("start_time", &self.start_time())?;
+        state.serialize_field("run_time", &self.run_time())?;
+        state.serialize_field("cpu_usage", &self.cpu_usage())?;
+        state.serialize_field("disk_usage", &self.disk_usage())?;
+        if let Some(uid) = self.user_id() {
+            let uid = *uid.deref();
+            state.serialize_field("user_id", &uid)?;
+        }
+        if let Some(gid) = self.group_id() {
+            let gid = *gid.deref();
+            state.serialize_field("group_id", &gid)?;
+        }
+        if let Some(pid) = self.session_id() {
+            state.serialize_field("session_id", &pid.as_u32())?;
+        }
+
+        state.end()
+    }
+}
+
+impl Serialize for dyn CpuExt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Cpu", 1)?;
+
+        state.serialize_field("cpu_usage", &self.cpu_usage())?;
+        state.serialize_field("name", &self.name())?;
+        state.serialize_field("vendor_id", &self.vendor_id())?;
+        state.serialize_field("brand", &self.brand())?;
+        state.serialize_field("frequency", &self.frequency())?;
+
+        state.end()
+    }
+}
+
+/*
+impl serde::Serialize for dyn SystemExt {
```

Comment:
For this one, I think it's better to not list `processes` as there are too many.